extractor/filesystem/os/pacman/pacman_test.go

```diff
@@ -0,0 +1,397 @@
+	"io/fs"
+	"os"
+	"path/filepath"
+	"reflect"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/osv-scalibr/extractor"
+	"github.com/google/osv-scalibr/extractor/filesystem"
+	"github.com/google/osv-scalibr/extractor/filesystem/internal/units"
+	"github.com/google/osv-scalibr/extractor/filesystem/os/pacman"
+	"github.com/google/osv-scalibr/extractor/filesystem/simplefileapi"
+	scalibrfs "github.com/google/osv-scalibr/fs"
+	"github.com/google/osv-scalibr/purl"
+	"github.com/google/osv-scalibr/stats"
+	"github.com/google/osv-scalibr/testing/fakefs"
+	"github.com/google/osv-scalibr/testing/testcollector"
+)
+
+func TestNew(t *testing.T) {
+	tests := []struct {
+		name    string
+		cfg     pacman.Config
+		wantCfg pacman.Config
+	}{
+		{
+			name: "default",
+			cfg:  pacman.DefaultConfig(),
+			wantCfg: pacman.Config{
+				MaxFileSizeBytes: 100 * units.MiB,
+			},
+		},
+		{
+			name: "custom",
+			cfg: pacman.Config{
+				MaxFileSizeBytes: 10,
+			},
+			wantCfg: pacman.Config{
+				MaxFileSizeBytes: 10,
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got := pacman.New(tt.cfg)
+			if !reflect.DeepEqual(got.Config(), tt.wantCfg) {
+				t.Errorf("New(%+v).Config(): got %+v, want %+v", tt.cfg, got.Config(), tt.wantCfg)
+			}
+		})
+	}
+}
+
+func TestFileRequired(t *testing.T) {
+	tests := []struct {
+		name             string
+		path             string
+		fileSizeBytes    int64
+		maxFileSizeBytes int64
+		wantRequired     bool
+		wantResultMetric stats.FileRequiredResult
+	}{
+		{
+			name:             "desc file",
+			path:             "var/lib/pacman/local/pacmanlinux-keyring-20241015-1/desc",
+			wantRequired:     true,
+			wantResultMetric: stats.FileRequiredResultOK,
+		},
+		{
+			name:             "desc file required if file size < max file size",
+			path:             "var/lib/pacman/local/argon2-20190702-6/desc",
+			fileSizeBytes:    100 * units.KiB,
+			maxFileSizeBytes: 1000 * units.KiB,
+			wantRequired:     true,
+			wantResultMetric: stats.FileRequiredResultOK,
+		},
+		{
+			name:             "desc file required if file size == max file size",
+			path:             "var/lib/pacman/local/audit-4.0.2-2/desc",
+			fileSizeBytes:    1000 * units.KiB,
+			maxFileSizeBytes: 1000 * units.KiB,
+			wantRequired:     true,
+			wantResultMetric: stats.FileRequiredResultOK,
+		},
+		{
+			name:             "desc file not required if file size > max file size",
+			path:             "var/lib/pacman/local/argon2-20190702-6/desc",
+			fileSizeBytes:    1000 * units.KiB,
+			maxFileSizeBytes: 100 * units.KiB,
+			wantRequired:     false,
+			wantResultMetric: stats.FileRequiredResultSizeLimitExceeded,
+		},
+		{
+			name:             "desc file required if max file size set to 0",
+			path:             "var/lib/pacman/local/audit-4.0.2-2/desc",
+			fileSizeBytes:    100 * units.KiB,
+			maxFileSizeBytes: 0,
+			wantRequired:     true,
+			wantResultMetric: stats.FileRequiredResultOK,
+		},
```

Comment:
Add some cases where we skip extraction because the file path isn't what the extractor wants